networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/discovery/discv5/DiscV5Service.java

```diff
@@ -121,15 +166,25 @@ public DiscV5Service(
         () -> discoverySystem.getBucketStats().getTotalLiveNodeCount());
   }
 
-  private NewAddressHandler maybeUpdateNodeRecord(
-      final boolean userExplicitlySetAdvertisedIpOrPort, final int advertisedTcpPort) {
-    if (userExplicitlySetAdvertisedIpOrPort) {
+  private NewAddressHandler maybeUpdateNodeRecord(final NetworkConfig p2pConfig) {
+    if (p2pConfig.hasUserExplicitlySetAdvertisedIps()) {
```

Comment:
I don't understand this, shouldn't it be inversed?

Reply:
Same behaviour as before. If user explicitly sets advertised IPs, they are added in the ENR. Otherwise, NewAddressHandler is communicated between nodes who don't have explicit advertised ips, in that case, we want to have a handler configured in discovery